### ISSUE TITLE
Plan to move from merlin 2 to merlin 3

### ISSUE THREAD

**Issue by hackwaly** [hackwaly/vscode-ocaml#102]

**Comment by kkirstein** [comment]
The current version of merlin (github/master, not yet released) now also compiles on Windows. So, from my point of view it would make sense to think about support of merlin 3. Unfortunately, the legacy mode of merlin 3 does not work with vscode-ocaml today. I have no experience with vscode extensions, but I can support with testing, especially on Windows.

**Comment by Khady** [comment]
> Unfortunately, the legacy mode of merlin 3 does not work with vscode-ocaml today.

Does not work with vscode-ocaml on windows you mean? I'm using vscode-ocaml and merlin 3 on linux since a few days without trouble.

**Comment by kkirstein** [comment]
@Khady Thanks for the feedback. Yes, I am using it on Windows (cygwin environment with mingw toolchain for ocaml). Merlin 3 works with vim, but not with vscode, so I concluded it to be a vscode issue. Maybe that was a hasty conclusion...
Did you specify any command line parameter via "ocaml.merlinPath"?

**Comment by Khady** [comment]
No, I have no special configuration. And I changed nothing in my vscode configuration when I upgraded to merlin 3.

Unfortunately I don't have a windows to try to reproduce the problem.

**Comment by hackwaly** [comment]
@kkirstein Yes. Merlin 3 says it support legacy protocol, but on windows it's buggy. Vim seems use new protocol, thus it works. see https://github.com/hackwaly/vscode-ocaml/issues/112#issuecomment-324861476
